pkg/systemd/hwinfo.jsx

```diff
@@ -24,61 +24,215 @@ import React from "react";
+    /* nosmt */
+    let promises = [cockpit.spawn(["lscpu"], { environ: ["LC_ALL=C.UTF-8"], }), cockpit.file("/proc/cmdline").read()];
+    return Promise.all(promises).then(values => {
+        let threads_per_core;
+        try {
+            threads_per_core = Number(values[0].split('\n')
+                    .find(l => l.indexOf('Thread(s) per core:') !== -1)
+                    .split(':')[1]);
+        } catch (e) {
+            console.warn(e);
+            return { available: false };
+        }
+        /* "nosmt" and "nosmt=force" are valid */
+        let nosmt_enabled = (values[1].indexOf("nosmt") !== -1 && values[1].indexOf("nosmt=") === -1) || values[1].indexOf("nosmt=force") !== -1;
+        /* available if threads>1 and the cmdline is valid */
+        let nosmt_available = threads_per_core > 1 && (values[1].indexOf("nosmt=") === -1 || values[1].indexOf("nosmt=force") !== -1);
+
+        cachedMitigations = { available: nosmt_available, nosmt_enabled: nosmt_enabled };
+        return cachedMitigations;
+    });
+};
+
+class CPUSecurityMitigationsDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.close = this.close.bind(this);
+        this.saveAndReboot = this.saveAndReboot.bind(this);
+        this.state = {
+            nosmt: undefined,
+            alert: undefined,
+            mitigationsAvailable: false,
+        };
+        AvailableMitigations().then(({ available, nosmt_enabled }) => {
+            this.setState({ mitigationsAvailable: available, nosmt: nosmt_enabled });
+        });
+    }
+
+    close() {
+        if (this.props.onClose)
+            this.props.onClose();
+    }
+
+    saveAndReboot() {
+        let options = [];
+        if (this.state.nosmt)
+            options = ['set', 'nosmt'];
+        else
+            options = ['remove', 'nosmt'];
+        cockpit.script(kernelopt_sh, options, { superuser: "require" })
+                .then(() => {
+                    cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: "require" })
+                            .catch(error => this.setState({ rebooting: false, alert: error }));
+                })
+                .catch(error => this.setState({ rebooting: false, alert: error }));
+    }
+
+    render() {
+        let rows = [];
+        if (this.state.nosmt !== undefined)
+            rows.push((
+                <ListView.Item key="nosmt" heading={ <span>{ _("Disable symmetric multithreading") } (nosmt)</span> }
+                               actions={ <div id="nosmt-switch">
+                                   <OnOffSwitch disabled={this.state.rebooting} onChange={ value => this.setState({ nosmt: value }) } state={ this.state.nosmt } />
+                               </div> } >
+                    <span>{ _("Read more at: ") }
+                        <a role="link" href="https://access.redhat.com/security/vulnerabilities/L1TF">https://access.redhat.com/security/vulnerabilities/L1TF</a>
```

Comment:
I can't open the link with a left click, it' seems the event gets eaten somewhere. Middle click works, but may be a bit  unobvious.

Reply:
This should also open in a new tab, even on left click. To do so, you need both target of _blank and a rel of noopener and noreferrer, like this:

```html
<a href="..." target="_blank" rel="noopener noreferrer">
```

I'm recommending that the link should open a new tab in browsers so people don't lose their work in Cockpit. This really should be the default for pretty much all outgoing links, as we're an app.

(BTW: The same issues exist with `window.open()` in JavaScript as with setting targets in HTML, and are solved in similar-ish ways.)

Reply:
We already set `Referrer-Policy": "no-referrer` globally, so you don't need that. (But of course it doesn't hurt really to repeat it)